### ISSUE TITLE
Cloudinary delayed image availability

### ISSUE THREAD

**Issue by rafaelcorreiapoli** [envisioning/techdb#69]
When the upload is finished and the image is ready on S3, we get a callback to finally ask it from Cloudinary. Cloudinary says the image is not found (404) and after some seconds it becomes available. That's strange since we are using Cloudinary Auto Upload feature, and it only depends if the image is available on the url we pass to it (in our case, S3)

**Comment by michellzappa** [comment]
Can we use a placeholder image while the final one loads?

**Comment by rafaelcorreiapoli** [comment]
Yes, but this behaviour should not be expected..also, the image is taking around 1 minute to become available, not acceptable. Another point is that it is only happening on production... I think is something related with cloudinary, i will do some tests with cloudImage to debug the problem.

**Comment by lnmunhoz** [comment]
I was getting this bug in production, but after a few tests it stops suddenly.

**Comment by rafaelcorreiapoli** [comment]
Yes, happened to me too and such instability from cloudinary is what concerns me...

**Comment by arthursoares** [comment]
So we monitor this?

**Comment by rafaelcorreiapoli** [comment]
I think I solved the Issue. I was using the uploaded callback, but this does not means that the image is already on S3. uploaded callback means that the image has arrived on our server, not on S3. There is a 'stored' callback that tells us that the image is stored on our adapters.
Using this callback, we guarantee that the image is available when Cloudinary asks for it.
By the way, I implemented a component called safeImg that displays a spinner while the final image has not loaded yet
